backend/de.metas.business/src/main/java/de/metas/bpartner/quick_input/service/BPartnerQuickInputService.java

```diff
@@ -424,36 +428,97 @@ private static BPartnerQuickInputId extractBpartnerQuickInputId(final @NonNull I
 
 	private BPartnerComposite toBPartnerComposite(@NonNull final I_C_BPartner_QuickInput template)
 	{
-		final BPGroupId groupId = BPGroupId.ofRepoIdOrNull(template.getC_BP_Group_ID());
-		if (groupId == null)
+		if (template.getC_BP_Group_ID() <= 0)
+		{
+			throw new FillMandatoryException(I_C_BPartner_QuickInput.COLUMNNAME_C_BP_Group_ID);
+		}
+
+		final ArrayList<BPartnerLocation> locations = getBPartnerLocations(template);
+		final BPartnerLocation uniqueLocationOfBPartnerTemplate = getUniqueLocationOfBPartnerTemplate(template);
+		if (uniqueLocationOfBPartnerTemplate != null)
 		{
-			throw new FillMandatoryException("C_BP_Group_ID");
+			locations.add(uniqueLocationOfBPartnerTemplate);
 		}
 
-		final LocationId existingLocationId = LocationId.ofRepoIdOrNull(template.getC_Location_ID());
-		if (existingLocationId == null)
+		if (locations.isEmpty())
 		{
-			throw new FillMandatoryException(I_C_BPartner_QuickInput.COLUMNNAME_C_Location_ID);
+			throw new FillMandatoryException(I_C_BPartner_QuickInput.COLUMNNAME_C_Location_ID, I_C_BPartner_Location_QuickInput.COLUMNNAME_C_BPartner_Location_QuickInput_ID);
 		}
 
-		final CountryId countryId = locationDAO.getCountryIdByLocationId(existingLocationId);
+		validatePricingSystemsForCountries(template, locations);
 
 		//
-		// Validate pricing setup
+		// BPartner (header)
+		final BPartner bpartner = getBPartner(template);
+
+		final ArrayList<BPartnerContact> contacts = getBPartnerContacts(template);
+
+		return BPartnerComposite.builder()
+				.orgId(OrgId.ofRepoId(template.getAD_Org_ID()))
+				.bpartner(bpartner)
+				.locations(locations)
+				.contacts(contacts)
+				.build();
+	}
+
+	private void validatePricingSystemsForCountries(final @NonNull I_C_BPartner_QuickInput template,
+
+			final ArrayList<BPartnerLocation> locations)
+	{
+
 		final PricingSystemId customerPricingSystemId = PricingSystemId.ofRepoIdOrNull(template.getM_PricingSystem_ID());
+		final PricingSystemId vendorPricingSystemId = PricingSystemId.ofRepoIdOrNull(template.getPO_PricingSystem_ID());
+
+		final ImmutableSet<CountryId> countryIds = locations.stream()
+				.map(BPartnerLocation::getExistingLocationId)
+				.map(locationDAO::getCountryIdByLocationId)
+				.collect(ImmutableSet.toImmutableSet());
+
+		//
+		// Validate pricing setup
 		if (customerPricingSystemId != null && template.isCustomer())
 		{
-			assertPriceListExists(customerPricingSystemId, countryId, SOTrx.SALES);
+			assertPriceListExists(customerPricingSystemId, countryIds, SOTrx.SALES);
```

Comment:
make sure only the address which is used in Sales Order is checked.
The other addresses IMHO shall be OK as is.

pls validate with @metas-mk  too.